app/controllers/nodes_controller.rb

```diff
@@ -0,0 +1,64 @@
+class NodesController < ApplicationController
+
+	def show
+		if params[:id]
+			@node = Node.find(params[:id])
+		else
+			@node = Node.root
+		end
+	end
+
+	def create
+		p = create_params
+		node = Node.create!(p)
+		redirect_to node_path(node.parent_id)
+	end
+
+	def edit
+		@node = Node.find(params[:id])
+	end
+
+	def update
+		@node = Node.find(params[:id])
+		if @node.update_attributes(node_params)
+			redirect_to node_path(@node.parent_id)
```

Comment:
In this case we can do a render of the show view (or an update view) instead of a redirect to the show method.